stellargraph/layer/deep_graph_infomax.py

```diff
@@ -72,27 +73,45 @@ class DeepGraphInfomax:
 
     Args:
         base_model: the base stellargraph model class
+        base_generator (Generator):
     """
 
-    def __init__(self, base_model):
+    def __init__(self, base_model, base_generator=None):
```

Comment:
Potentially we want to pass the `CorruptedGenerator` here, but I feel passing the base model's generator has a nice symmetry here.

Reply:
I'm inclined to go with `CorruptedGenerator ` here because most (all?) models are constructed with their training generator, also `CorruptedGenerator` will have the advantage that the check for `CorruptibleGenerator` has already been done.